parts/linux/cloud-init/artifacts/cse_main.sh

```diff
@@ -321,7 +321,7 @@ if ! [[ ${API_SERVER_NAME} =~ ^[0-9]+\.[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
        API_SERVER_DNS_RETRIES=200
     fi
     if [[ "${ENABLE_HOSTS_CONFIG_AGENT}" != "true" ]]; then
-        RES=$(retrycmd_if_failure ${API_SERVER_DNS_RETRIES} 1 10 nslookup ${API_SERVER_NAME})
+        RES=$(logs_to_events "AKS.CSE.apiserverNslookup" "retrycmd_if_failure ${API_SERVER_DNS_RETRIES} 1 10 nslookup ${API_SERVER_NAME}")
```

Comment:
hm. is STS=$? even doing anything here?

Reply:
basically worried about a similar exit/return issue like we saw before

if cse logs in ab e2e / kusto look good im fine w it

Reply:
yes surprisingly